src/ansys/mechanical/core/embedding/background.py

```diff
@@ -82,6 +82,12 @@ def post(self, callable: typing.Callable):
             raise RuntimeError("Cannot use BackgroundApp after stopping it.")
         return BackgroundApp.__poster.post(callable)
 
+    def try_post(self, callable: typing.Callable):
```

Comment:
minor - you can factor the check into a method used by both `post` and `try_post`